### PR TITLE
Note timing side channels

### LINKED ISSUE #400: Hard to imagine ECH handling in constant time - ponder and document?
Given all the complexity and options it's hard to imaging ECH handling ever being contstant-time. It'd probably be no harm to ponder that at least and maybe add some security considerations text before we're done.

### PR DESCRIPTION
**Pull request by chris-wood**
Closes #400 

There may be other ways to spell this, or other places to put it, so please feel free to leave suggestions or propose alternative PRs if desired.

cc @dennisjackson, @davidben, @cjpatton

### CODE REVIEW

File: draft-ietf-tls-esni.md [hunk 1]
```diff
@@ -1305,14 +1307,22 @@ send context-specific values in ClientHelloOuter.
...
 signaling unusual behavior to passive observers, provided the choice of value
 and value itself are not sensitive. See {{dont-stick-out}}.
 
+## Inner ClientHello {#inner-clienthello}
+
+Values which dependent on the contents of ClientHelloInner, such as the
+true server name, can influence how client-facing servers process this message.
+In particular, timing side channels can reveal information about the contents
+of ClientHelloInner. Implementations should take such side channels into
+consideration when reasoning about the privacy properties that ECH provides.
```
Comment (davidben) [inline draft-ietf-tls-esni.md:1324]: https://www.rfc-editor.org/rfc/rfc6919.html#section-2 ;-)

**Review by davidben — APPROVED** [review]

File: draft-ietf-tls-esni.md (hunk 1 above)
Comment (davidben) [reply draft-ietf-tls-esni.md:1324]: (I don't have a suggestion and am not requesting a change. Just amused.)

Comment (chris-wood) [reply draft-ietf-tls-esni.md:1324]: Heh, yeah. I mostly just copied boilerplate language we use in Privacy Pass. 🤷

File: draft-ietf-tls-esni.md [hunk 2]
```diff
@@ -1305,14 +1307,22 @@ send context-specific values in ClientHelloOuter.
...
+about which extensions are compressed, so inner extensions which only sometimes
+match the corresponding outer extension SHOULD NOT be compressed.
 
 Clients MAY include additional extensions in ClientHelloOuter to avoid
 signaling unusual behavior to passive observers, provided the choice of value
 and value itself are not sensitive. See {{dont-stick-out}}.
 
+## Inner ClientHello {#inner-clienthello}
+
+Values which dependent on the contents of ClientHelloInner, such as the
```
Comment (klinvill) [inline draft-ietf-tls-esni.md:1320]: Minor typo:
> Values which _depend_ on ...

**Review by dennisjackson — APPROVED** [review]

**Pull request merged**